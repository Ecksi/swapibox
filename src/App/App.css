.mainArea {
  display: flex;
  overflow: scroll;
}

.scrollWrap {
  width: 37%;
  margin-top: 15%;
  overflow: hidden;
}

.loadingWrap {
  width: 63%;
}