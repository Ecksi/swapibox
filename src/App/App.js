import React, { Component } from 'react';
import CardContainer from '../CardContainer/CardContainer';
import TextCrawl from '../TextCrawl/TextCrawl';
import { fetchPeople, fetchPlanets, fetchVehicles } from '../ApiCalls/ApiCalls';
import './App.css';

class App extends Component {
  constructor (props) {
    super(props);
    this.state = {
      people: '',
      planets: '',
      vehicles: '',
      favorites: []
    };
   
    this.fetchData = this.fetchData.bind(this);
  }

  fetchData (event) {
    event.preventDefault();
    if (event.target.innerText === 'People') {
      return this.peopleData();
    } else if (event.target.innerText === 'Planets') {
      return this.planetData();
    } else if (event.target.innerText === 'Vehicles') {
      return this.vehicleData();
    } else if (event.target.innerText.split(' ')[0] === 'Favorites') {
      return this.showFavorites();
    }
  }

  async peopleData () {
    this.setState({
      people: await fetchPeople(),
      planets: '',
      vehicles: ''
    });
  }

  async planetData () {
    this.setState({
      people: '',
      planets: await fetchPlanets(),
      vehicles: ''
    });
  }

  async vehicleData () {
    this.setState({
      people: '',
      planets: '',
      vehicles: await fetchVehicles()
    });
  }

  showFavorites = () => {
    this.setState({
      people: '',
      planets: '',
      vehicles: ''
    });
  }

  addToFavorites = (card) => {
    this.setState({
      favorites: [card, ...this.state.favorites]
    });
  }

  removeFromFavorites = (card) => {
    const getIndex = this.state.favorites.findIndex(favorite => favorite.id === card.id);

    this.state.favorites.splice(getIndex, 1);

    this.setState ({
      favorites: [...this.state.favorites]
    });
  }

  render() {
    return (
      <div className="App">
        <button onClick={this.fetchData}>People</button>
        <button onClick={this.fetchData}>Planets</button>
        <button onClick={this.fetchData}>Vehicles</button>
        <button onClick={this.fetchData}>Favorites {this.state.favorites.length}</button>
        <CardContainer 
          people={this.state.people}
          planets={this.state.planets}
          vehicles={this.state.vehicles}
          favorites={this.state.favorites}
          addFavorite={this.addToFavorites}
          removeFavorite={this.removeFromFavorites}
        />
        <TextCrawl />
      </div>
    );
  }
}

export default App;
