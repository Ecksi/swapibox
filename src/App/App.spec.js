import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  let app;

  beforeEach(() => {
    app = shallow(<App />, { disableLifecycleMethods: true });
  });

  it('should match the snapshot', () => {
    expect(app).toMatchSnapshot();
  });

  it('should have a default state of an empty array for people', () => {
    expect(app.state('people')).toEqual([]);
  });

  it('should have a default state of an empty array for planets', () => {
    expect(app.state('planets')).toEqual([]);
  });

  it('should have a default state of an empty array for vehicles', () => {
    expect(app.state('vehicles')).toEqual([]);
  });

  it('should have a default state of an empty array for favorites', () => {
    expect(app.state('favorites')).toEqual([]);
  });

  it('should have a default state of false for loading', () => {
    expect(app.state('loading')).toEqual(false);
  });

  it('should have a default state of an empty string for loadScreen', () => {
    expect(app.state('loadScreen')).toEqual('');
  });

  it('should fetch the appropiate data when fetchData is called', {

  });

  it('should set the state of loading to true when fetchData is called', () => {

  });

  // the fooData's pattern should be 1 function, easier to test / more dynamic
  it('should set the state of loading to false when peopleData is called', () => {

  });

  it('should set the state of people when peopleData is called', async () => {
    const mockPeopleData = jest.fn();
    await app.mockPeopleData;


    expect(mockPeopleData).toEqual({});
  });

  it('should set the state of planets when planetData is called', async () => {

  });

  it('should set the state of vehicles when vehicleData is called', async () => {

  });

  it('should reset the states of people, planets, and vehicles when favorites is called', async () => {

  });

  it('should add a card to the state of favorites when called', () => {

  });


});