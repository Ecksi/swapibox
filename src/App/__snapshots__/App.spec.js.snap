// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <ButtonContainer
          favoritesCounter={Array []}
          fetchData={[Function]}
/>,
        <main
          className="mainArea"
>
          <div
                    className="scrollWrap"
          >
                    <TextCrawl />
          </div>
          <div
                    className="loadingWrap"
          >
                    <CardContainer
                              addFavorite={[Function]}
                              favorites={Array []}
                              people={Array []}
                              planets={Array []}
                              removeFavorite={[Function]}
                              vehicles={Array []}
                    />
          </div>
</main>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "favoritesCounter": Array [],
          "fetchData": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="scrollWrap"
>
              <TextCrawl />
</div>,
            <div
              className="loadingWrap"
>
              <CardContainer
                            addFavorite={[Function]}
                            favorites={Array []}
                            people={Array []}
                            planets={Array []}
                            removeFavorite={[Function]}
                            vehicles={Array []}
              />
</div>,
          ],
          "className": "mainArea",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <TextCrawl />,
              "className": "scrollWrap",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <CardContainer
                addFavorite={[Function]}
                favorites={Array []}
                people={Array []}
                planets={Array []}
                removeFavorite={[Function]}
                vehicles={Array []}
/>,
              "className": "loadingWrap",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "addFavorite": [Function],
                "favorites": Array [],
                "people": Array [],
                "planets": Array [],
                "removeFavorite": [Function],
                "vehicles": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "main",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <ButtonContainer
            favoritesCounter={Array []}
            fetchData={[Function]}
/>,
          <main
            className="mainArea"
>
            <div
                        className="scrollWrap"
            >
                        <TextCrawl />
            </div>
            <div
                        className="loadingWrap"
            >
                        <CardContainer
                                    addFavorite={[Function]}
                                    favorites={Array []}
                                    people={Array []}
                                    planets={Array []}
                                    removeFavorite={[Function]}
                                    vehicles={Array []}
                        />
            </div>
</main>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "favoritesCounter": Array [],
            "fetchData": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="scrollWrap"
>
                <TextCrawl />
</div>,
              <div
                className="loadingWrap"
>
                <CardContainer
                                addFavorite={[Function]}
                                favorites={Array []}
                                people={Array []}
                                planets={Array []}
                                removeFavorite={[Function]}
                                vehicles={Array []}
                />
</div>,
            ],
            "className": "mainArea",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <TextCrawl />,
                "className": "scrollWrap",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <CardContainer
                  addFavorite={[Function]}
                  favorites={Array []}
                  people={Array []}
                  planets={Array []}
                  removeFavorite={[Function]}
                  vehicles={Array []}
/>,
                "className": "loadingWrap",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "addFavorite": [Function],
                  "favorites": Array [],
                  "people": Array [],
                  "planets": Array [],
                  "removeFavorite": [Function],
                  "vehicles": Array [],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "main",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
